### PR TITLE
Feat: Filter by Rating, Price Range, stock, and clear Filter

### PR DESCRIPTION
**Pull request by pritam-kr**
- [Added three more filters by Category, Price Slider, Rating](https://github.com/pritam-kr/Pearl/commit/ee8471569fe62f825431670cae31facd317f1b77)
 
#### And also update the state  on view
 - [Update Filter component ](https://github.com/pritam-kr/Pearl/commit/1df77e9e7a6fdc4185d072581eb55d6a845b1fa6)
 
 -  [Filter by Rating, Price range, stock JSX file](https://github.com/pritam-kr/Pearl/commit/00b579e45230143b691357c7aabe2bbe3c42dff7)

https://user-images.githubusercontent.com/84632214/158520238-64133c68-6a87-42ca-8c60-08976f8077f9.mp4

### CODE REVIEW

File: src/Context/GlobalContext/StateContext.jsx [hunk 1]
```diff
@@ -1,72 +1,83 @@
...
-
+import {
+    createContext,
+    useContext,
+    useReducer,
+    useEffect,
+    useState,
+} from "react";
+import axios from "axios";
+import { uniqueCategory } from "../../utils/uniqueCategory";
```
Comment (AbhishekkGautam) [inline src/Context/GlobalContext/StateContext.jsx:9]: We can use barrel export here. Export all methods from index.js inside the utils folder.

File: src/Context/GlobalContext/StateReducer.js [hunk 2]
```diff
@@ -47,6 +50,35 @@ export const stateReducerFun = (state, action) => {
                     categoryName: [...state.filters.categoryName, action.payload],
                 },
             };
+
+            case STAR_RATING: 
+            
+            if(action.payload === 4){
+                return{...state, filters: {...state.filters, rating: action.payload}}
```
Comment (AbhishekkGautam) [inline src/Context/GlobalContext/StateReducer.js]: We can pass payload to rating without writing 4 if else. 
if( payload !== "" ) {
return rating: payload 
} 
return rating:null

File: src/Context/GlobalContext/StateReducer.js [hunk 3]
```diff
@@ -47,6 +50,35 @@ export const stateReducerFun = (state, action) => {
...
+
+            case PRICE_RANGE:
+            
+            return {...state, filters: {...state.filters, priceRange: action.payload}}
+
+            case CLEAR_FILTER:
+
+            return {
+               ...state, filters: {...state.filters,  sortBy: "",
+               priceRange: 500,
```
Comment (AbhishekkGautam) [inline src/Context/GlobalContext/StateReducer.js]: Suggestion: we can create a method which returns the max price among all the products. That way we don't need to worry about the range of price.

File: src/component/Filter/Filter.jsx [hunk 4]
```diff
@@ -39,23 +38,24 @@ const Filter = () => {
...
           </h6>
           <input
             className="input range-input"
             id="price-range"
             type="range"
-            min="200"
-            max="1000"
-            step="50"
+            min="2000"
+            max="66000"
```
Comment (AbhishekkGautam) [inline src/component/Filter/Filter.jsx]: These price attributes are hard coded. We should pass the state variable max={max_price}.

**Review by AbhishekkGautam — APPROVED** [review]
Rest everything looks good to me.

**Review by manojasarsa — APPROVED** [review]
The code looks good and clean.
Use Barrel Exports to rollup exports from several modules into a single convenient module. (from utils folder)
Rest looks good apart from what @AbhishekkGautam suggested.

File: src/Context/GlobalContext/StateReducer.js (hunk 2 above)
Comment (pritam-kr) [reply src/Context/GlobalContext/StateReducer.js]: I updated this.

File: src/Context/GlobalContext/StateReducer.js (hunk 3 above)
Comment (pritam-kr) [reply src/Context/GlobalContext/StateReducer.js]: Done

File: src/component/Filter/Filter.jsx (hunk 4 above)
Comment (pritam-kr) [reply src/component/Filter/Filter.jsx]: Done

**Pull request merged**